### PR TITLE
Introduce "strip" and deprecate "environments" option

### LINKED ISSUE #41: Replace "environments" option with "strip" option
I'd like to propose replacing our current [`environment` option](https://github.com/simplabs/ember-test-selectors#options) with a simple boolean `strip` option.

This option should default to the opposite of the [`tests` property](https://github.com/ember-cli/ember-cli/blob/v2.10.0/lib/broccoli/ember-app.js#L156) of the `EmberApp` instance, which is `true` when either `ember test` was used or when we're not in production mode.

To recreate the old behavior you'd simply have to set `strip` to `EmberApp.env() === 'production'`.

/cc @marcoow @pangratz

### PR DESCRIPTION
**Pull request by Turbo87**
Resolves #41

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -4,19 +4,38 @@
...
+
+    var appOptions = app.options || {};
     var addonOptions = appOptions['ember-test-selectors'] || {};
-    var environments = addonOptions.environments || ['production'];
 
-    if (environments.indexOf(registry.app.env) !== -1) {
-      var StripTestSelectorsTransform = require('./strip-test-selectors');
+    if (addonOptions.environments) {
+      ui.writeDeprecateLine('The "environments" option in "ember-test-selectors" has been replaced ' +
+        'with a "strip" option. Use e.g. "strip: EmberApp.env() === \'production\'" instead to ' +
```
Comment (marcoow) [inline index.js]: _"with __the__ "strip" option"_?

Comment (Turbo87) [inline index.js]: my gut feeling says "with a "strip" option" sounds better, but I'm happy with either. maybe we should ask a native english speaker. 🤷‍♂️

Comment (marcoow) [inline index.js]: > we should ask a native english speaker

👍 /cc @oliverbarnes

Comment (sapeien) [inline index.js]: *the* would be more correct in my opinion, because it matches the *the* from the beginning of the sentence.

Comment (chriskrycho) [inline index.js]: Concur about `the` over `a`.

Comment (Turbo87) [inline index.js]: Updated. Thanks everyone!

**Pull request merged**